googlepubsub/src/main/scala/akka/stream/alpakka/googlepubsub/scaladsl/GooglePubSub.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.googlepubsub.scaladsl
+
+import java.security.PrivateKey
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.stream.alpakka.googlepubsub._
+import akka.stream.scaladsl.{Flow, Keep, Sink, Source}
+import akka.{Done, NotUsed}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object GooglePubSub extends GooglePubSub {
+  private[googlepubsub] override val httpApi = HttpApi
+  private[googlepubsub] def getSession(clientEmail: String, privateKey: PrivateKey): Session =
+    new Session(clientEmail, privateKey)
+}
+
+protected[googlepubsub] trait GooglePubSub {
+  private[googlepubsub] def httpApi: HttpApi
+  private[googlepubsub] def getSession(clientEmail: String, privateKey: PrivateKey): Session
+
+  private implicit def matToEx(implicit mat: Materializer): ExecutionContext = mat.executionContext
+
+  /**
+   * Creates a flow to that publish messages to a topic and emits the message ids
+   */
+  def publish(projectId: String,
+              apiKey: String,
+              clientEmail: String,
+              privateKey: PrivateKey,
+              topic: String,
+              parallelism: Int = 1)(implicit actorSystem: ActorSystem,
```

Comment:
I think its enough to just require `materializer` - where are you using the system?